nemo/collections/tts/data/text_to_speech_dataset_lhotse.py

```diff
@@ -0,0 +1,317 @@
+        tokenizer_config=None,
+        load_16khz_audio: bool = True,
+        use_text_conditioning_tokenizer: bool = False,
+        pad_context_text_to_max_duration: bool = False,
+        context_duration_min: float = 3.0,
+        context_duration_max: float = 10.0
+    ):
+        super().__init__()
+        self.sample_rate = sample_rate
+        self.text_tokenizer = None
+        self.align_prior_hop_length = align_prior_hop_length
+        self.volume_norm = volume_norm
+
+        self.bos_id = bos_id
+        self.eos_id = eos_id
+        self.audio_bos_id = audio_bos_id
+        self.audio_eos_id = audio_eos_id
+        self.codec_model_downsample_factor = codec_model_downsample_factor
+        self.include_align_prior = prior_scaling_factor is not None
+        self.prior_scaling_factor = prior_scaling_factor
+        self.load_cached_codes_if_available = load_cached_codes_if_available
+        self.dataset_type = dataset_type
+        self.tokenizer_config = tokenizer_config
+        self.load_16khz_audio = load_16khz_audio
+        self.use_text_conditioning_tokenizer = use_text_conditioning_tokenizer
+        self.text_conditioning_tokenizer = None
+        self.pad_context_text_to_max_duration = pad_context_text_to_max_duration
+        self.context_duration_min = context_duration_min
+        self.context_duration_max = context_duration_max
+
+    def __getitem__(self, cuts):
+        cuts = cuts.sort_by_duration()
+
+        logging.debug(f"Len: {len(cuts)}")
+
+        # load audios and text
+        num_codec_frames = []
+        align_priors = []
+        context_audios = []
+        context_audios_lens = []
+        target_audios = []
+        target_audios_lens = []
+        target_audios_16khz = []
+        target_audios_16khz_lens = []
+        context_text_tokens = []
+        context_text_tokens_lens = []
+        has_text_context_list = []
+        target_text_tokens = []
+        target_text_tokens_lens = []
+
+        for i, cut in enumerate(cuts):
+            # load target/answer audio
+            answer_audio = torch.FloatTensor(cut.target_audio.resample(self.sample_rate).load_audio()).squeeze(0)
+            if self.volume_norm:
+                answer_audio = normalize_volume_torch(answer_audio)
+
+            answer_audio = torch.nn.functional.pad(
+                answer_audio,
+                (0, self.codec_model_downsample_factor - (answer_audio.shape[0] % self.codec_model_downsample_factor)),
+                value=0
+            ).unsqueeze(0)
+
+            answer_audio_len = answer_audio.shape[1]
+            target_audios.append(answer_audio)
+            target_audios_lens.append(answer_audio_len)
+            num_frames = int(answer_audio_len / self.codec_model_downsample_factor) + 1 # +1 for EOS
+            num_codec_frames.append(num_frames)
+
+            # load context audio
+            context_audio = torch.FloatTensor(cut.resample(self.sample_rate).load_audio()).squeeze(0)
```

Comment:
What audio is this loading? Is it different from target?

Reply:
It is the context audio/speaker reference/most similar audio that you created for all metafiles and shared with me.